py-src/ltmain.py

```diff
@@ -22,84 +22,93 @@
 threads = []
 currentClient = 0
 
-class Printer():
-  cur = ""
 
-  def write(self, text):
-    self.cur += text
-    if text == "\n":
-      self.flush()
+class Printer(object):
+    cur = ""
 
-  def flush(self):
-    send(currentClient, "editor.eval.python.print", {"msg": self.cur, "file": name})
-    self.cur = ""
+    def write(self, text):
+        self.cur += text
+        if text == "\n":
+            self.flush()
 
-  def readlines(self):
-    return None
+    def flush(self):
+        send(currentClient, "editor.eval.python.print",
+             {"msg": self.cur, "file": name})
+        self.cur = ""
+
+    def readlines(self):
+        return None
+
+    def read(self):
+        return None
 
-  def read(self):
-    return None
 
 def asUnicode(s):
-  try:
-    return unicode(s)
-  except:
-    return str(s)
+    try:
+        return unicode(s)
+    except NameError:  # python 3
+        return str(s)
+
 
 def ensureUtf(s):
-  if type(s) == unicode:
-    return s.encode('utf8', 'ignore')
-  else:
-    return str(s)
+    try:
+        return s.decode('utf8', 'ignore')
```

Comment:
@kenny-evitt I included that here. It depends whether ensureUtf is supposed to "convert to utf" or "show as utf". If it's convert then this should be decode (python2 is very forgiving/lax).